packages/git/src/browser/history/git-history-widget.ts

```diff
@@ -119,6 +109,21 @@ export class GitHistoryWidget extends GitBaseWidget<GitHistoryListNode> implemen
         }
     }
 
+    protected async addFileChangeNodesToCommit(commit: GitCommitNode) {
+        for (const fileChange of commit.fileChanges) {
```

Comment:
using a for loop with await will make it that the promises are resolved one after the other. Using `commit.fileChnges.map(..)` would allow to run the promises "in parallel", in that if they are doing a server roundtrip multiple requests can be made before they are responded.